processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -341,14 +352,37 @@ public String getImportName() {
     }
 
     /**
-     * Whether this type is imported by means of an import statement in the currently generated source file (meaning it
-     * can be referenced in the generated source using its simple name) or not (meaning it has to be referenced using
-     * the fully-qualified name).
+     * Whether this type is to be imported by means of an import statement in the currently generated source file
+     * (it can be referenced in the generated source using its simple name) or not (referenced using the FQN).
      *
      * @return {@code true} if the type is imported, {@code false} otherwise.
      */
-    public boolean isImported() {
-        return isImported;
+    public boolean isToBeImported() {
+        if ( isToBeImported == null ) {
+            String trimmedName = trimSimpleClassName( name );
+            if ( notToBeImportedTypes.containsKey( trimmedName ) ) {
+                return false;
+            }
+            String trimmedQualifiedName = trimSimpleClassName( qualifiedName );
+            String importedType = toBeImportedTypes.get( trimmedName );
+
+            isToBeImported = false;
```

Comment:
Shouldn't we initialize this value when we return false on line 164 as well?

Reply:
164?

Reply:
364. Sorry a typo